tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -533,12 +521,34 @@ def do_rolling_bounce(self, processor, counter, current_generation):
 
                     # version probing should trigger second rebalance
-                    current_generation = current_generation + 1
-
-                    for p in self.processors:
-                        monitors[p].wait_until("Successfully joined group with generation " + str(current_generation),
-                                               timeout_sec=60,
-                                               err_msg="Never saw output 'Successfully joined group with generation " + str(current_generation) + "' on" + str(p.node.account))
+                    # now we check that after consecutive rebalances we have synchronized generation
+                    generation_synchronized = False
+                    retries = 0
+
+                    while retries < 10:
+                        processor_found = list(node.account.ssh_capture("grep \"Successfully joined group with generation\" %s | awk \'{for(i=1;i<=NF;i++) {if ($i == \"generation\") beginning=i+1; if($i== \"(org.apache.kafka.clients.consumer.internals.AbstractCoordinator)\") ending=i }; for (j=beginning;j<ending;j++) printf $j; printf \"\\n\"}\'" % processor.LOG_FILE, allow_fail=True))
+                        first_other_processor_found = list(first_other_node.account.ssh_capture("grep \"Successfully joined group with generation\" %s | awk \'{for(i=1;i<=NF;i++) {if ($i == \"generation\") beginning=i+1; if($i== \"(org.apache.kafka.clients.consumer.internals.AbstractCoordinator)\") ending=i }; for (j=beginning;j<ending;j++) printf $j; printf \"\\n\"}\'" % first_other_processor.LOG_FILE, allow_fail=True))
+                        second_other_processor_found = list(second_other_node.account.ssh_capture("grep \"Successfully joined group with generation\" %s | awk \'{for(i=1;i<=NF;i++) {if ($i == \"generation\") beginning=i+1; if($i== \"(org.apache.kafka.clients.consumer.internals.AbstractCoordinator)\") ending=i }; for (j=beginning;j<ending;j++) printf $j; printf \"\\n\"}\'" % second_other_processor.LOG_FILE, allow_fail=True))
+
+                        if len(processor_found) > 0 and len(first_other_processor_found) > 0 and len(second_other_processor_found) > 0:
+                            self.logger.info("processor: " + str(processor_found))
+                            self.logger.info("first other processor: " + str(first_other_processor_found))
+                            self.logger.info("second other processor: " + str(second_other_processor_found))
+
+                            processor_generation = int(processor_found[len(processor_found) - 1].strip().split()[0])
+                            first_other_processor_generation = int(first_other_processor_found[len(first_other_processor_found) - 1].strip().split()[0])
+                            second_other_processor_generation = int(second_other_processor_found[len(second_other_processor_found) - 1].strip().split()[0])
+
```

Comment:
same here have a `retrieve_generation_num` method for extracting the generation number from captured output